Remotion/ObjectBinding/UnitTests/Design/DropDownEditorBaseTest.cs

```diff
@@ -18,142 +18,161 @@
+    private Mock<EditorControlBase> _mockEditorControlBase;
 
     [SetUp]
     public void SetUp ()
     {
-      _mockRepository = new MockRepository();
-
-      _mockTypeDescriptorContext = _mockRepository.StrictMock<ITypeDescriptorContext>();
-      _mockServiceProvider = _mockRepository.StrictMock<IServiceProvider>();
-      _mockWindowsFormsEditorService = _mockRepository.StrictMock<IWindowsFormsEditorService>();
-      _mockDropDownEditorBase = _mockRepository.PartialMock<MockDropDownEditorBase>();
-      _mockEditorControlBase = _mockRepository.PartialMock<EditorControlBase> (_mockServiceProvider, _mockWindowsFormsEditorService);
+      _mockTypeDescriptorContext = new Mock<ITypeDescriptorContext> (MockBehavior.Strict);
+      _mockServiceProvider = new Mock<IServiceProvider> (MockBehavior.Strict);
+      _mockWindowsFormsEditorService = new Mock<IWindowsFormsEditorService> (MockBehavior.Strict);
+      _mockDropDownEditorBase = new Mock<MockDropDownEditorBase>() { CallBase = true };
+      _mockEditorControlBase = new Mock<EditorControlBase> (_mockServiceProvider.Object, _mockWindowsFormsEditorService.Object) { CallBase = true };
     }
 
     [Test]
     public void EditValue ()
     {
-      Expect.Call (_mockTypeDescriptorContext.Instance).Return (new object());
-      Expect.Call (_mockServiceProvider.GetService (typeof (IWindowsFormsEditorService))).Return (_mockWindowsFormsEditorService);
-      Expect.Call (_mockDropDownEditorBase.NewCreateEditorControl (_mockTypeDescriptorContext, _mockWindowsFormsEditorService))
-          .Return (_mockEditorControlBase);
-      using (_mockRepository.Ordered())
-      {
-        _mockEditorControlBase.Value = "The input value";
-        _mockWindowsFormsEditorService.DropDownControl (_mockEditorControlBase);
-        Expect.Call (_mockEditorControlBase.Value).Return ("The output value");
-      }
-      _mockRepository.ReplayAll();
-
-      object actual = _mockDropDownEditorBase.EditValue (_mockTypeDescriptorContext, _mockServiceProvider, "The input value");
-
-      _mockRepository.VerifyAll();
+      _mockTypeDescriptorContext.Setup (_ => _.Instance).Returns (new object()).Verifiable();
+      _mockServiceProvider.Setup (_ => _.GetService (typeof (IWindowsFormsEditorService))).Returns (_mockWindowsFormsEditorService.Object).Verifiable();
+      _mockDropDownEditorBase.Setup (_ => _.NewCreateEditorControl (_mockTypeDescriptorContext.Object, _mockWindowsFormsEditorService.Object))
+          .Returns (_mockEditorControlBase.Object)
+          .Verifiable();
+      var sequence = new MockSequence();
+      _mockEditorControlBase.SetupProperty (_ => _.Value);
+      _mockEditorControlBase.Object.Value = "The input value";
+      _mockWindowsFormsEditorService.Setup (_ => _.DropDownControl (_mockEditorControlBase.Object));
+      _mockEditorControlBase.InSequence (sequence).Setup (_ => _.Value).Returns ("The output value").Verifiable();
```

Comment:
```suggestion
      _mockEditorControlBase.Object.Value = "The input value";
      _mockWindowsFormsEditorService.InSequence (sequence).Setup (_ => _.DropDownControl (_mockEditorControlBase.Object));
      _mockEditorControlBase.InSequence (sequence).Setup (_ => _.Value).Returns ("The output value").Verifiable();
```

What's the syntax to cover line 56 with Sequence?

Reply:
`_mockEditorControlBase.InSequence (sequence).Setup (_ => _.Value).Returns ("The input value");`

Reply:
okay, good. is this something we can cover in the rewriter?

Reply:
ok, it's harder than I initially thought. The code I posted above causes the test to fail. In `Rhino.Mocks` it checks if `_mockEditorControlBase.Value` is assigned a value. The same could be done in `Moq` with `mock.SetupSet`, but this throws a `System.NotSupportedException`, because the return type of the member is not mockable. I need to take a closer look at this.🧐

Reply:
please timebox and I don't understand "the return type of the member is not mockable"

Reply:
It throws an exception with the message "The return type of the member is not mockable", but i don't know why this happens

Reply:
jep, strange indeed.

Reply:
Ok, I don't know what causes the error, however I don't think that the missing `InSequence` causes the test to falsely pass. What is your opinion on this?

Reply:
Hmm...make Value use property behavior and a callback-check to the next call and verify that the Value is correct?